src/mon/OSDMonitor.cc

```diff
@@ -796,14 +796,20 @@ class OSDUtilizationDumper : public CrushTreeDumper::Dumper<F> {
 
   bool get_osd_utilization(int id, int64_t* kb, int64_t* kb_used,
 			   int64_t* kb_avail) const {
+    // we always return true for OSDs as the specific OSDs can be out,
```

Comment:
> could you be specific why checking osdmap.is_out(id) in get_bucket_utilization() does not work, or is a bad idea?

@tchaikov  That's not true. I think the `is_out()` method shall work too(It is my first at all solution actually:-). But I have the following concerns too:

- I think https://github.com/ceph/ceph/pull/10308/files?diff=unified#diff-0a5db46a44ae9900e226289a810f10e8R803 is unreachable, as we have done sanity check at:

```
  void dump_stray(F *f) {
    for (int i = 0; i < osdmap->get_max_osd(); i++) {
      if (osdmap->exists(i) && !this->is_touched(i))
	    dump_item(CrushTreeDumper::Item(i, 0, 0), f);
    }
  }
```

- I am afraid there is other possibilities that an OSD report zeroed stats, please check:

```
void OSDService::update_osd_stat(vector<int>& hb_peers)
{
  Mutex::Locker lock(stat_lock);

  osd_stat.hb_in.swap(hb_peers);
  osd_stat.hb_out.clear();

  osd->op_tracker.get_age_ms_histogram(&osd_stat.op_queue_age_hist);

  // fill in osd stats too
  struct store_statfs_t stbuf;
  int r = osd->store->statfs(&stbuf);
  if (r < 0) {
    derr << "statfs() failed: " << cpp_strerror(r) << dendl;